lib/init/features/database.js

```diff
@@ -1,17 +1,41 @@
 'use strict';
 
 var chalk = require('chalk');
-
+var api = require('../../api');
 var prompt = require('../../prompt');
 var logger = require('../../logger');
+var configstore = require('../../configstore');
+var utils = require('../../utils');
+var fsutils = require('../../fsutils');
+var RSVP = require('rsvp');
+
+var _getConsoleDBRules = function(projectId) {
+  return api.request('GET', '/.settings/rules.json', {
+    auth: true,
+    origin: utils.addSubdomain(api.realtimeOrigin, projectId)
+  }).then(function(response) {
+    return JSON.stringify(response.body, null, 2);
+  });
+};
+
+var _writeDBRules = function(projectId, fileName, config) {
+  return _getConsoleDBRules(projectId).then(function(rules) {
+    return config.writeProjectFile(fileName, rules);
+  }).then(function() {
+    utils.logSuccess('Database Rules for ' + projectId + ' have been downloaded to ' + fileName + ' .');
+    logger.info('Future modifications to ' + fileName + ' will update Database Rules when you do');
+    logger.info(chalk.bold('firebase deploy') + '.');
+  });
+};
 
 module.exports = function(setup, config) {
   setup.config.database = {};
+  var projectId = configstore.get('activeProjects')[config.projectDir];
```

Comment:
We don't need to and shouldn't be getting the `projectId` from configstore here -- instead, it should be made available on `setup` as a contextual variable based on the choice the user just made in a previous prompt. This is also how we should get the instance name. The `projects` response payload from [this line](https://github.com/firebase/firebase-tools/blob/ll-msg-rules-deploy/lib/init/features/project.js#L23) contains a structure something like this:

    {myprojectid: {instances: {database: ["instance-for-myprojectid"]}}

So to get the instance, you'd just need to do `projects[projectId].instances.databases[0]`. Once the dev has picked a project id, let's just remember both the projectId and the instance to `setup` and then access it later from the Database init.

Make sense?